### ISSUE TITLE
SC1073 parse error in double quoted string.  (possible improvement the message is NOT incorrect as issued in this case)

### ISSUE THREAD

**Issue by dshuman52** [koalaman/shellcheck#3085]
#### For bugs
- Rule Id (if any, e.g. SC1000):
- My shellcheck version (`shellcheck --version` or "online"):
- ShellCheck - shell script analysis tool
version: 0.9.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net

- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
b[1]='2'

a="${b(c])"


```

#### Here's what shellcheck currently says:  (the online version not my version associated with Geany)

$ shellcheck myscript
 
[Line 4:](javascript:setPosition(4, 4))
a="${b(c])"
   ^-- [SC1009](https://www.shellcheck.net/wiki/SC1009) (info): The mentioned syntax error was in this parameter expansion.
          ^-- [SC1073](https://www.shellcheck.net/wiki/SC1073) (error): Couldn't parse this double quoted string. Fix to allow more checks.
 
[Line 5:](javascript:setPosition(5, 1))

^-- [SC1072](https://www.shellcheck.net/wiki/SC1072) (error): Expected end of double quoted string. Fix any mentioned problems and try again.

$

#### Here's what I wanted or expected to see:

Clearly in this case the only double quoted string is closed.  Clearly also inside this double quoted string there are unpaired [] or () in this case.  The only ( should obviously be a [.  Admittedly in this case the message is NOT incorrect; however it is also not very helpful in more complicated situations.  Question would it be preferable when finding paired double quotes in this case that mismatched pairs of delimiters inside the double quotes would potentially provide a more helpful identification of the issue (maybe two messages one for the mismatched bracket/parens too.  I have had some lengthy strings of code and identification of the mismatched bracket, paren, curly bracket, sq bracket would in my opinion be an improvement.  You obviously have more experience in this area and maybe this is not a better idea.

I use Geany to call shellcheck.  Geany does a wonderful job of visually flagging mismatched double and single quotes, in most cases.  Such that I fix them before invoking shellcheck.  Therefore considering that the editor is performing the quote pairing check in a reasonably in a visual manner.  In my limited opinion having shellcheck more concerned with mismatched parens and brackets would be an improvement as the quote problems are often effectively illustrated by Geany.   I am NOT advocating that checking of quotes by shellcheck should  be eliminated.

I have not figured out how to demo this yet but I got a similar situation generating an SC1078 message about mismatched double quotes that often in my case turn out to be mismatched parens or brackets.  As I tend to construct multiple variable strings primarily for use by echo it is not unusual for me to fat finger one or more bracket/parens in the multi variable strings being created. I even used these multi-variable strings to construct validated paths and file names, etc.  Knowing that the Double Quote errors even mismatched double quotes could be a problem with parens and brackets helps me now.  But when coding late night if mismatched brackets and parens were more actively identified, I believe it could be helpful.

Please consider the appropriate changes based on your better knowledge of any adverse effects the request could create.
